panel_segmentation/lidar/pcd_data.py

```diff
@@ -0,0 +1,238 @@
+            filtered_points = []
+            final_point_record = laspy.PackedPointRecord.empty(
+                f.header.point_format)
+            # Get crs of LiDAR data
+            self.source_crs = f.header.parse_crs()
+            # Check if crs has multiple components
+            if self.source_crs.is_compound:
+                # Get horizontal crs
+                # Only need horizontal projection since Shapely polygon is 2D
+                horizontal_crs = self.source_crs.sub_crs_list[0]
+                dst_crs = f"EPSG:{horizontal_crs.to_epsg()}"
+            else:
+                dst_crs = f"EPSG:{self.source_crs.to_epsg()}"
+            # Make coordinate transformer from EPSG:4326 to LiDAR crs
+            self.transformer = Transformer.from_crs(
+                "EPSG:4326", dst_crs, always_xy=True)
+            # For shapely polygons
+            if self.polygon:
+                # Project polygons in latitude, longitude (in "EPSG:4326")
+                # onto lidar coordinate system
+                projected_poly = transform(
+                    self.transformer.transform, self.polygon)
+            # For latitude, longitude coordinate point
+            elif self.latitude and self.longitude:
+                # Project latitude, longitude coordinate points onto lidar
+                # coordinate system
+                x_mid, y_mid = self.transformer.transform(
+                    self.longitude, self.latitude)
+                # Make bbox with x_mid, y_mid as it's centers
+                bbox_size = lat_lon_bbox_size / 2
+                left, right = x_mid - bbox_size, x_mid + bbox_size
+                bottom, top = y_mid - bbox_size, y_mid + bbox_size
+            # Get [x, y] scale factors to match transformed lat long to
+            # array values
+            self.scales = f.header.scales
+            self.offsets = f.header.offsets
+            # Read values by chunks to avoid crashing
+            for points in f.chunk_iterator(CHUNK_SIZE):
+                # Resize coordinates with lidar's scale and offset
+                x_coord = points["X"] * self.scales[0] + self.offsets[0]
+                y_coord = points["Y"] * self.scales[1] + self.offsets[1]
+                # Create bbox
+                if self.polygon:
+                    bbox = shapely.contains_xy(
+                        projected_poly, x_coord, y_coord)
+                elif self.latitude and self.longitude:
+                    bbox = ((x_coord >= left) & (x_coord <= right) &
+                            (y_coord >= bottom) & (y_coord <= top))
+                # Filter out unnecessary classification and get only
+                # unclassified or building classification
+                # Classification info at:
+                # https://www.usgs.gov/ngp-standards-and-specifications/lidar-base-specification-tables
+                filter_class = (points.classification == 1) | (
+                    points.classification == 6)
```

Comment:
we should make the classification a passable parameter, and also document what these numbers mean